Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import Foundation
+import SwiftUI
+
+extension OfflineMapAreasView {
+    /// The model class for the offline map areas view.
+    @MainActor
+    class MapViewModel: ObservableObject {
+        /// The portal item ID of the web map.
+        private let portalItemID: String?
+        
+        /// The offline map task.
+        private let offlineMapTask: OfflineMapTask
+        
+        /// The preplanned offline map information.
+        @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
+        
+        /// A Boolean value indicating whether the map has preplanned map areas.
+        @Published private(set) var hasPreplannedMapAreas = false
+        
+        /// A Boolean value indicating whether the user has authorized notifications to be shown.
+        var canShowNotifications = false
```

Comment:
I believe this needs to be `@Published` since the model changes it's value.
```suggestion
        @Published var canShowNotifications = false
```